pkg/util/parse/parse.go

```diff
@@ -0,0 +1,398 @@
+			DeviceModelRef: &v12.LocalObjectReference{
+				Name: "modbus-sample-model",
+			},
+			Protocol: v1alpha2.ProtocolConfig{
+				Modbus: &v1alpha2.ProtocolConfigModbus{SlaveID: pointer.Int64Ptr(1)},
+				Common: &v1alpha2.ProtocolConfigCommon{
+					TCP: &v1alpha2.ProtocolConfigTCP{
+						IP:   "10.222.1.1",
+						Port: 502,
+					},
+				},
+			},
+			PropertyVisitors: []v1alpha2.DevicePropertyVisitor{
+				{
+					PropertyName: "property0",
+					ReportCycle:  5000,
+					CollectCycle: 5000,
+					VisitorConfig: v1alpha2.VisitorConfig{
+						Modbus: &v1alpha2.VisitorConfigModbus{
+							Register:       "HoldingRegister",
+							Offset:         pointer.Int64Ptr(0),
+							Limit:          pointer.Int64Ptr(1),
+							Scale:          1,
+							IsSwap:         true,
+							IsRegisterSwap: true,
+						},
+					},
+				},
+				{
+					PropertyName: "property1",
+					ReportCycle:  5000,
+					CollectCycle: 5000,
+					VisitorConfig: v1alpha2.VisitorConfig{
+						Modbus: &v1alpha2.VisitorConfigModbus{
+							Register:       "HoldingRegister",
+							Offset:         pointer.Int64Ptr(1),
+							Limit:          pointer.Int64Ptr(1),
+							Scale:          1,
+							IsSwap:         true,
+							IsRegisterSwap: true,
+						},
+					},
+				},
+				{
+					PropertyName: "property2",
+					ReportCycle:  5000,
+					CollectCycle: 5000,
+					VisitorConfig: v1alpha2.VisitorConfig{
+						Modbus: &v1alpha2.VisitorConfigModbus{
+							Register:       "HoldingRegister",
+							Offset:         pointer.Int64Ptr(2),
+							Limit:          pointer.Int64Ptr(1),
+							Scale:          1,
+							IsSwap:         true,
+							IsRegisterSwap: true,
+						},
+					},
+				},
+			},
+			Data: v1alpha2.DeviceData{},
+			NodeSelector: &v12.NodeSelector{
+				NodeSelectorTerms: []v12.NodeSelectorTerm{
+					{
+						MatchExpressions: []v12.NodeSelectorRequirement{
+							{
+								Key:      "",
+								Operator: v12.NodeSelectorOpIn,
+								Values:   []string{"test"},
+							},
+						},
+					},
+				},
+			},
+		},
+		Status: v1alpha2.DeviceStatus{
+			Twins: []v1alpha2.Twin{
+				{
+					PropertyName: "property0",
+					Desired: v1alpha2.TwinProperty{
+						Value: "0",
+						Metadata: map[string]string{
+							"timestamp": "1550049403598",
+							"type":      "int",
+						},
+					},
+					Reported: v1alpha2.TwinProperty{},
+				},
+				{
+					PropertyName: "property1",
+					Desired: v1alpha2.TwinProperty{
+						Value: "0",
+						Metadata: map[string]string{
+							"timestamp": "1550049403598",
+							"type":      "int",
+						},
+					},
+					Reported: v1alpha2.TwinProperty{},
+				},
+				{
+					PropertyName: "property2",
+					Desired: v1alpha2.TwinProperty{
+						Value: "0",
+						Metadata: map[string]string{
+							"timestamp": "1550049403598",
+							"type":      "int",
+						},
+					},
+					Reported: v1alpha2.TwinProperty{},
+				},
+			},
+		},
+	},
+}
+
+var tmpDeviceModels = []v1alpha2.DeviceModel{
```

Comment:
is this `tmpDeviceModels` never used? we can remove it.

Reply:
done